cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/queue/JobQueueDataset.java

```diff
@@ -179,16 +180,7 @@ private boolean isTriggerSatisfied(Trigger trigger, List<Notification> notificat
       // is initially created
```

Comment:
no, because `TimeTrigger#isSatisfied` should only be called when it's inside a composite trigger. This is for backward compatibility, since `TimeTrigger#isSatisfied` looks for the filed cron expression in notification, so that multiple time triggers in a composite can be distinguished. This field does not exist in pre-4.3, so if there are unprocessed notifications from pre-4.3 after upgrade, the notification won't satisfy `TimeTrigger#isSatisfied`. We can also let `TimeTrigger#isSatisfied` distinguish whether the time trigger is inside composite trigger or not, but that will be more complicated